test/schrodinger/rdkit_extensions/test_monomer_coordgen.cpp

```diff
@@ -372,3 +373,147 @@ BOOST_DATA_TEST_CASE(
                                      << "\nExpected coords: " << expected_str
                                      << "\nComputed coords: " << actual_str);
 }
+
+BOOST_AUTO_TEST_SUITE(TestMonomerCoordgenCheckCoords)
+
+static RDKit::RWMol
+make_molecule_with_coords(const std::vector<std::pair<double, double>>& xy,
+                          const std::vector<std::pair<int, int>>& bonds = {})
+{
+    auto mol = RDKit::RWMol();
+
+    // Add atoms
+    for (size_t i = 0; i < xy.size(); ++i)
+        mol.addAtom(new RDKit::Atom(), false, false); // no extra processing
```

Comment:
I think you might be leaking atoms which is why memtest is failing

mol.addAtom(new RDKit::Atom(), false, true); // true = take ownership
https://github.com/rdkit/rdkit/blob/9a4b3e2fc6d4573b4fbd272ddc4f7af4307158a8/Code/GraphMol/ROMol.cpp#L382